src/algorithm/good_thomas_algorithm.rs

```diff
@@ -9,26 +11,22 @@ use algorithm::FFTAlgorithm;
 pub struct GoodThomasAlgorithm<T> {
     width: usize,
     // width_inverse: usize,
-    width_size_fft: Box<FFTAlgorithm<T>>,
+    width_size_fft: Rc<FFTAlgorithm<T>>,
 
     height: usize,
     // height_inverse: usize,
-    height_size_fft: Box<FFTAlgorithm<T>>,
+    height_size_fft: Rc<FFTAlgorithm<T>>,
```

Comment:
Because if you have something like Box<Trait>, that's what's called a trait object: https://doc.rust-lang.org/book/trait-objects.html

Trait objects are very useful because they allow for virtual calls, and virtual calls are what allow us to not know (and not care) whether our FFTAlgorithm instance is MixedRadix or Radix4. But trait objects absolutely must sit behind references or owned pointers.

Owned pointers means Rc or box, which means heap allocation. And references w/ stack allocation means complex lifetime management, and I don't think it's possible to design a library that does this ergonomically.

So for practical purposes, we can't use trait objects without heap allocation. Sad :(

This means that a sub-FFT cannot hide which type of FFT it is from its parent, and at some point, someone needs to hold a stack variable with a concrete, known-at-compile-time type. And it's going to be a crazy type like MixedRadix<MixedRadix<RadersAlgorithm<Radix4>,MixedRadix<Butterfly8, Butterfly7>>, RadersAlgorithm<MixedRadix<Radix4, Butterfly5>>>

Summary: We currently "hide" the type of FFT by putting it behind a trait object, but trait objects for all intents and purposes require heap allocation, so we can't use them if we want to avoid heap allocation.